staging/src/k8s.io/kube-openapi/pkg/aggregator/aggregator.go

```diff
@@ -0,0 +1,480 @@
+			}
+		}
+		return ref
+	}, sp)
+
+	// Remove unused definitions
+	orgDefinitions := sp.Definitions
+	sp.Definitions = spec.Definitions{}
+	for k, v := range orgDefinitions {
+		if usedDefinitions[k] {
+			sp.Definitions[k] = v
+		}
+	}
+}
+
+func equalSchemaMap(s1, s2 map[string]spec.Schema) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for k, v := range s1 {
+		v2, found := s2[k]
+		if !found {
+			return false
+		}
+		if !EqualSchema(&v, &v2) {
+			return false
+		}
+	}
+	return true
+}
+
+func equalSchemaArray(s1, s2 []spec.Schema) bool {
+	if s1 == nil || s2 == nil {
+		return s1 == nil && s2 == nil
+	}
+	if len(s1) != len(s2) {
+		return false
+	}
+	for _, v1 := range s1 {
+		found := false
+		for _, v2 := range s2 {
+			if EqualSchema(&v1, &v2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	for _, v2 := range s2 {
+		found := false
+		for _, v1 := range s1 {
+			if EqualSchema(&v1, &v2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+func equalSchemaOrBool(s1, s2 *spec.SchemaOrBool) bool {
+	if s1 == nil || s2 == nil {
+		return s1 == s2
+	}
+	if s1.Allows != s2.Allows {
+		return false
+	}
+	if !EqualSchema(s1.Schema, s2.Schema) {
+		return false
+	}
+	return true
+}
+
+func equalSchemaOrArray(s1, s2 *spec.SchemaOrArray) bool {
+	if s1 == nil || s2 == nil {
+		return s1 == s2
+	}
+	if !EqualSchema(s1.Schema, s2.Schema) {
+		return false
+	}
+	if !equalSchemaArray(s1.Schemas, s2.Schemas) {
+		return false
+	}
+	return true
+}
+
+func equalStringArray(s1, s2 []string) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for _, v1 := range s1 {
+		found := false
+		for _, v2 := range s2 {
+			if v1 == v2 {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	for _, v2 := range s2 {
+		found := false
+		for _, v1 := range s1 {
+			if v1 == v2 {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+func equalFloatPointer(s1, s2 *float64) bool {
+	if s1 == nil || s2 == nil {
+		return s1 == s2
+	}
+	return *s1 == *s2
+}
+
+func equalIntPointer(s1, s2 *int64) bool {
+	if s1 == nil || s2 == nil {
+		return s1 == s2
+	}
+	return *s1 == *s2
+}
+
+// EqualSchema returns true if models have the same properties and references
+// even if they have different documentation.
+func EqualSchema(s1, s2 *spec.Schema) bool {
+	if s1 == nil || s2 == nil {
+		return s1 == s2
+	}
+	if s1.Ref.String() != s2.Ref.String() {
+		return false
+	}
+	if !equalSchemaMap(s1.Definitions, s2.Definitions) {
+		return false
+	}
+	if !equalSchemaMap(s1.Properties, s2.Properties) {
+		fmt.Println("Not equal props")
+		return false
+	}
+	if !equalSchemaMap(s1.PatternProperties, s2.PatternProperties) {
+		return false
+	}
+	if !equalSchemaArray(s1.AllOf, s2.AllOf) {
+		return false
+	}
+	if !equalSchemaArray(s1.AnyOf, s2.AnyOf) {
+		return false
+	}
+	if !equalSchemaArray(s1.OneOf, s2.OneOf) {
```

Comment:
Sorry, I am not sure if I understand this comment. Do you mean `{oneOf: [x]}` and `{oneOf: [x,x]}` should be considered equal?